zubhub_frontend/zubhub/src/views/create_activity/create_activity_step1.jsx

```diff
@@ -0,0 +1,101 @@
+import React, {useState, useEffect} from 'react'
+import clsx from 'clsx';
+import 'react-toastify/dist/ReactToastify.css';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Grid,
+} from '@material-ui/core';
+import { vars, handleTextFieldChange, handleTextFieldBlur} from '../create_project/createProjectScripts';
+import CustomButton from '../../components/button/Button';
+import projectStyles from '../../assets/js/styles/views/create_project/createProjectStyles';
+import { styles } from '../../assets/js/styles/views/create_activity/createActivityStyles'
+import commonStyles from '../../assets/js/styles';
+import InputText from '../../components/inputText/inputText';
+import FormLabel from '../../components/form_labels/formLabel';
+import Input from '../../components/input/input';
+const useProjectStyles = makeStyles(projectStyles);
+const useStyles = makeStyles(styles);
+const useCommonStyles = makeStyles(commonStyles);
+
+function CreateActivityStep1(props) {
+    const classes = useProjectStyles()
+    const activity_classes = useStyles()
+    const common_classes = useCommonStyles()
+    const [newActivity, setNewActivity] = useState(props.newActivity)
+    const {t} = {...props} 
+    const setFieldValue = (value, field) => {
+      const fieldObject = {}
+      fieldObject[field] = value
+      setNewActivity((prevactivity) => ({ ...prevactivity, ...fieldObject }))
+    }
+    
+  return (
+    <div className={activity_classes.createActivityStepContainer}>
+          <Grid container spacing={3}>
+            <Grid item xs={12} className={common_classes.marginTop1em}>
+              
+                <FormLabel 
+                  label={"title"}
+                  required= {true}
+                  classes={classes}
+                  common_classes={common_classes}
+                  inputOrder={1}
+                  fieldLabel={t('createActivity.inputs.title.label')}
+                />
+                <Input 
+                  classes= {classes}
+                  defaultValue={props.values['title']? props.values.title : ''}
+                  label={'title'}
+                  {...props}
```

Comment:
please do not send all the props to child components. Only send the required props

Reply:
OK I will change it , I did for the UploadFile component and i will limit the parameters to the required ones for all the others as well
🙏